corehq/apps/hqwebapp/signals.py

```diff
@@ -26,9 +27,27 @@ def clear_failed_logins_and_unlock_account(sender, request, user, **kwargs):
 
 
 @receiver(user_login_failed)
-def add_failed_attempt(sender, credentials, **kwargs):
+def add_failed_attempt(sender, credentials, token_failure=False, **kwargs):
     user = CouchUser.get_by_username(credentials['username'])
-    if user and not user.is_locked_out() and user.supports_lockout():
+    if not user:
+        metrics_counter('commcare.auth.invalid_user')
+        return
+
+    if token_failure:
+        metrics_counter('commcare.auth.invalid_token')
+
+    locked_out = user.is_locked_out()
+
+    if locked_out:
+        lockout_result = 'locked_out'
+    else:
+        lockout_result = 'should_be_locked_out' if user.should_be_locked_out() else 'allowed_to_retry'
+
+    metrics_counter('commcare.auth.failed_attempts', tags={
+        'result': lockout_result
+    })
+
+    if not locked_out:
```

Comment:
changing this condition has caused some ResourceConflicts. See https://github.com/dimagi/commcare-hq/pull/28939/files#r566692930

Reply:
This appears to be the result of a quickcache race condition, demonstrated [here](https://github.com/dimagi/commcare-hq/pull/29087). The reason this actually happens pretty regularly is because the document submission process submits multiple documents quickly (either document + submission log or document + document retry). Forcing the metric logging code to refetch a user won't eliminate the possibility of a stale document (another node could still create the same race condition), but should make these conflicts much, much less common